File: PILA CON ARREGLOS/pilaarreglos.cs
using System;

public class Pila
{
    private const int TAMANIO_MAXIMO = 100;
    private int[] pila;
    private int tope;
    
    public Pila()
    {
        pila = new int[TAMANIO_MAXIMO];
        tope = -1;
    }
    
    public void Apilar(int numero)
    {
        if (tope == TAMANIO_MAXIMO - 1)
        {
            Console.WriteLine("Desbordamiento de Pila");
            return;
        }
        pila[++tope] = numero;
    }
    
    public int Desapilar()
    {
        if (tope == -1)
        {
            Console.WriteLine("Subdesbordamiento de Pila");
            return -1;
        }
        return pila[tope--];
    }
    
    public int Mirar()
    {
        if (tope == -1)
        {
            Console.WriteLine("La pila está vacía");
            return -1;
        }
        return pila[tope];
    }
    
    public bool EstaVacia()
    {
        return tope == -1;
    }
    
    public bool EstaLlena()
    {
        return tope == TAMANIO_MAXIMO - 1;
    }
    
    public static void Main(string[] args)
    {
        Pila p = new Pila();
        p.Apilar(10);
        p.Apilar(20);
        p.Apilar(30);
        
        Console.WriteLine("Elemento Superior: " + p.Mirar());
        Console.WriteLine("Extrae elemento: " + p.Desapilar());
        Console.WriteLine("Elemento Superior: " + p.Mirar());
    }
}

File: PILA CON ARREGLOS/pilaarreglos.js
class Pila {
    constructor() {
        this.TAMANIO_MAXIMO = 100;
        this.pila = new Array(this.TAMANIO_MAXIMO);
        this.tope = -1;
    }
    
    apilar(numero) {
        if (this.tope === this.TAMANIO_MAXIMO - 1) {
            console.log("Desbordamiento de Pila");
            return;
        }
        this.tope++;
        this.pila[this.tope] = numero;
    }
    
    desapilar() {
        if (this.tope === -1) {
            console.log("Subdesbordamiento de Pila");
            return -1;
        }
        const elemento = this.pila[this.tope];
        this.tope--;
        return elemento;
    }
    
    mirar() {
        if (this.tope === -1) {
            console.log("La pila está vacía");
            return -1;
        }
        return this.pila[this.tope];
    }
    
    estaVacia() {
        return this.tope === -1;
    }
    
    estaLlena() {
        return this.tope === this.TAMANIO_MAXIMO - 1;
    }
}

// Ejemplo de uso
const p = new Pila();
p.apilar(10);
p.apilar(20);
p.apilar(30);

console.log("Elemento Superior:", p.mirar());
console.log("Extrae elemento:", p.desapilar());
console.log("Elemento Superior:", p.mirar());